devtools/dump_devinfo.py

```diff
@@ -452,9 +481,196 @@ async def get_smart_fixture(device: SmartDevice, batch_size: int):
     sw_version = sw_version.split(" ", maxsplit=1)[0]
 
     save_filename = f"{model}_{hw_version}_{sw_version}.json"
-    copy_folder = "kasa/tests/fixtures/smart/"
+    copy_folder = SMART_FOLDER
     return save_filename, copy_folder, final
 
 
+@cli.command(name="generate-supported")
+async def generate_supported():
+    """Generate the SUPPORTED.md from the fixtures."""
+    supported = {
+        PLUGS: {"kasa": {}, "tapo": {}},
+        POWER_STRIPS: {"kasa": {}, "tapo": {}},
+        WALL_SWITCHES: {"kasa": {}, "tapo": {}},
+        BULBS: {"kasa": {}, "tapo": {}},
+        LIGHT_STRIPS: {"kasa": {}, "tapo": {}},
+    }
+    _get_iot_supported(supported)
+    _get_smart_supported(supported)
+
+    _update_supported_file(SUPPORTED_FILENAME, supported, False)
+    _update_supported_file(README_FILENAME, supported, True)
+
+
+def _update_supported_file(filename, supported, summary):
+    with open(filename) as f:
+        contents = f.readlines()
+
+    start_index = end_index = None
+    for index, line in enumerate(contents):
+        if line == "<!--SUPPORTED_START-->\n":
+            start_index = index + 1
+        if line == "<!--SUPPORTED_END-->\n":
+            end_index = index
```

Comment:
Instead of doing something like this, perhaps we should use some templating engine like jinja (https://realpython.com/primer-on-jinja-templating/) which is used to generate the real files? This would make changing the structure much easier w/o need to touch the generation code.

Reply:
I took a look into this and for now I'd prefer to keep it simple so that anyone can follow the current simple fixture instructions:

> please open a pull request to update the list and add a fixture file (`use python -m devtools.dump_devinfo` to generate one).

Otherwise we'll have to add instructions about installing python-kasa with extras which might deter some folks.

Also at the moment the `.md` files can be edited and previewed in the IDE without having to run them through a templating engine.  Perhaps we could look at having the `SUPPORTED.md` templated into an rst as part of the docs generation in a subsequent PR but AFAICT the README is a bit more troublesome as github expects it to be in the source and prevents `include` directives.

Reply:
I would be very much more in favor of making it easier to maintain and modify the template over having to instruct someone how to install a single package (which we can also include in the error output if we cannot import the template lib?).

Also, I wouldn't mind too much about having to install some extras being a deterrent; someone who wants to help and contribute will do so, and `dev-dependencies` is a common practise.

Whatever we decide to do, it's worth adding comments inline to indicate that some specific parts shall not be manually modified but are autogenerated.

Reply:
I've added that comment and I've also simplified the generation a bit to use `string.Template` although inline in the python rather than embedded in the README.